#                               Introduction

* Video games are popular all over the world. They are enjoyed by all ages. Video game industry is huge and the spending on video games per year is huge too. Sales of different types of games vary widely between countries due to local preferences.





In [12]:
#Importing libraries
import pandas as pd
import plotly.express as px
import plotly.graph_objects as go
import warnings
warnings.filterwarnings("ignore")
print("The modules are imported")

#Importing dataset
df= pd.read_csv('./Data/vgsales.csv')
df.head()


The modules are imported


Rank                      Name Platform    Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37

In [13]:
df.shape

(16598, 11)

* En utilisant la fonction df.shape, nous voyons que notre jeu de données a 16598 lignes et 11 colonnes.

In [14]:
df.isnull().sum() 

Rank              0
Name              0
Platform          0
Year            271
Genre             0
Publisher        58
NA_Sales          0
EU_Sales          0
JP_Sales          0
Other_Sales       0
Global_Sales      0
dtype: int64

nous chercherons les valeurs manquantes en utilisant la fonction df.isnull().sum(). Nous observons que la colonne Year a 271 valeurs manquantes et la colonne Publisher a 58 valeurs manquantes

# 1. Quelle région a le mieux performé en termes de ventes ?
Nous utiliserons les ventes moyennes réalisées par région et comparerons les résultats. Avant de faire cela, assurons-nous de savoir comment calculer la moyenne. Nous utiliserons une ligne de code très simple pour cela, c'est-à-dire df['Region'].mean(), où 'region' représente les différentes régions de l'ensemble de données et la fonction mean() est utilisée pour calculer la moyenne. Nous observons que notre sortie arrive en décimales, pour convertir les valeurs en millions, multiplions le résultat par 10,00,000. Le code final devrait ressembler à ceci.

In [15]:
x=(df['NA_Sales'].mean()*1000000)
y=(df['EU_Sales'].mean()*1000000)
z=(df['JP_Sales'].mean()*1000000)
q=(df['Other_Sales'].mean()*1000000)
p=(df['Global_Sales'].mean()*1000000)

print("The average sales in North America =", (f"${x:,.3f}")) #comma separated values till 3 decimal place and $ sign
print("The average sales in Europe =",(f"${y:,.3f}"))
print("The average sales in Japan =",(f"${z:,.3f}"))
print("The average sales in other regions =",(f"${q:,.3f}"))
print("The average sales globally =",(f"${p:,.3f}"))

The average sales in North America = $264,667.430
The average sales in Europe = $146,652.006
The average sales in Japan = $77,781.660
The average sales in other regions = $48,063.020
The average sales globally = $537,440.656


In [16]:
colors = ['lightslategray',] * 4
colors[1]='darkgray'
colors[2]='grey'
colors[3]='dimgrey'
colors[0] = 'crimson'

bar1 = go.Figure(data=[go.Bar(
    y=['Global','North America', 'Europe', 'Japan',
       'Other'],
    x=[537440.656,264667.430, 146652.006, 77781.660, 48063.020],
    orientation='h',
    marker_color=colors # marker color can be a single color value or an iterable
)])
bar1.update_layout(title_text='Region with highest sales on an average')
bar1.update_xaxes(title='Average Sales')
bar1.update_yaxes(title='Regions')

Nous avons utilisé avec succès un graphique à barres pour démontrer les résultats. Il est maintenant clair quelle région a réalisé le plus de ventes (attendez-vous à des ventes mondiales) en termes de jeux vidéo. Nous observons que l'**Amérique du Nord** enregistre les ventes moyennes les plus élevées de **264 667,430 USD**. Il est clair maintenant que nous pouvons utiliser le graphique à barres pour répondre à des questions telles que, quel pays vend le plus de jeux vidéo au monde ? Cela nous aidera à faire quelques changements de stratégie.

Le résultat ci-dessus répond également à notre 8ème question, c'est-à-dire : ** Y a-t-il une région qui a dépassé la moyenne mondiale des ventes ?**

La réponse est non'**. Aucune région n'a dépassé les ventes moyennes mondiales. Les ventes mondiales moyennes sont de **537 440,656 millions de dollars.**

# 2. Les meilleures consoles de jeu sont Microsoft (Xbox), Sony (Playstation) et Nintendo, Google agissant comme un nouveau concurrent.



Le monde devient de plus en plus connecté chaque jour, et de plus en plus de tendances et de préférences en matière de jeux émergent dans le monde entier.


Nous utiliserons la fonction df.groupby de pandas pour notre analyse.

**groupby() : cette fonction est utilisée pour regrouper ou combiner de grandes quantités de données et d'opérations de calcul sur ces groupes.**


* La base de données **données** que nous avons créée nous fournira les ventes totales (globales, NA, UE, JP, autres) basées sur différentes plateformes. La trame de données ressemblera à ceci :

In [17]:
#Grouping the north america sales based on each platform
data2 = pd.DataFrame(df.groupby("Platform")[["NA_Sales"]].sum().sort_values(by=['NA_Sales'],ascending=[False]).reset_index())
data2.rename(columns = {'Platform':'Platform_NA'}, inplace = True)

#Grouping the europe sales based on each platform
data3 = pd.DataFrame(df.groupby("Platform")[["EU_Sales"]].sum().sort_values(by=['EU_Sales'],ascending=[False]).reset_index())
data3.rename(columns = {'Platform':'Platform_EU'}, inplace = True)

#Grouping the japan sales based on each platform
data4 = pd.DataFrame(df.groupby("Platform")[["JP_Sales"]].sum().sort_values(by=['JP_Sales'],ascending=[False]).reset_index())
data4.rename(columns = {'Platform':'Platform_JP'}, inplace = True)

#Grouping the other region sales based on each platform
data5 = pd.DataFrame(df.groupby("Platform")[["Other_Sales"]].sum().sort_values(by=['Other_Sales'],ascending=[False]).reset_index())
data5.rename(columns = {'Platform':'Platform_other'}, inplace = True)

#Concatenating our datasets
data=pd.concat([data2,data3,data4,data5],axis=1)
data.head(3)

Platform_NA  NA_Sales Platform_EU  EU_Sales Platform_JP  JP_Sales  \
0        X360    601.05         PS3    343.71          DS    175.57   
1         PS2    583.84         PS2    339.29          PS    139.82   
2         Wii    507.71        X360    280.58         PS2    139.20   

  Platform_other  Other_Sales  
0            PS2       193.44  
1            PS3       141.93  
2           X360        85.54

* Nous réalisons que la trame de données seule rendra très difficile l'analyse du résultat. Par conséquent, nous traçons le résultat à l'aide d'un graphique linéaire.

In [18]:
from plotly.subplots import make_subplots #import new library

subplot1 = make_subplots(rows=4, cols=1, shared_yaxes=True,subplot_titles=("North American top platforms","Europe top platforms","Japan top platforms","Other regions top platforms"))

#Subplot for North America
subplot1.add_trace(go.Bar(x=data['Platform_NA'], y=data['NA_Sales'],
                    marker=dict(color=[1, 2, 3],coloraxis="coloraxis")),1, 1)

#Subplot for Europe
subplot1.add_trace(go.Bar(x=data['Platform_EU'], y=data['EU_Sales'],
                    marker=dict(color=[4, 5, 6], coloraxis="coloraxis")),                         2, 1)
                   
#Subplot for Japan
subplot1.add_trace(go.Bar(x=data['Platform_JP'], y=data['JP_Sales'],
                    marker=dict(color=[7, 8, 9], coloraxis="coloraxis")),
                    3, 1)

##Subplot for Other Regions
subplot1.add_trace(go.Bar(x=data['Platform_other'], y=data['Other_Sales'],
                    marker=dict(color=[10, 11, 12], coloraxis="coloraxis")),
                   4, 1)
                   
subplot1.update_layout(height=900,width=500,coloraxis=dict(colorscale='Magenta'), showlegend=False)
subplot1.show()

Le graphique nous montre les principales plates-formes préférées des utilisateurs dans différentes régions. Nous observons ce qui suit :

* **X360** (Microsoft) est la console préférée des utilisateurs en Amérique du Nord, avec un total de 601,05 millions de dollars.
* **PS3** (Sony) est la console préférée des utilisateurs en Europe avec un total de 343,71 millions de dollars.
* **DS** (Nintendo) est la console préférée des utilisateurs au Japon, avec un total de 175,57 millions de dollars.
* **PS2** (Sony) est la console préférée des utilisateurs d'autres régions, avec un total de 193,44 millions de dollars.

* Cela implique que notre hypothèse concernant Sony, Nintendo et Microsoft parmi les meilleures consoles était correcte.

# 3. Quels sont les 10 jeux les plus vendus actuellement dans le monde ?
* Nous utiliserons une approche similaire en regroupant les jeux par rapport aux ventes mondiales et en observant les 10 meilleurs jeux.

In [19]:
top = pd.DataFrame(df.groupby("Name")[["Global_Sales"]].sum().sort_values(by=['Global_Sales'],ascending=[False]).reset_index())
top.head(10) #Printing the top 10 results

Name  Global_Sales
0                      Wii Sports         82.74
1              Grand Theft Auto V         55.92
2               Super Mario Bros.         45.31
3                          Tetris         35.84
4                  Mario Kart Wii         35.82
5               Wii Sports Resort         33.00
6        Pokemon Red/Pokemon Blue         31.37
7         Call of Duty: Black Ops         31.03
8  Call of Duty: Modern Warfare 3         30.83
9           New Super Mario Bros.         30.01

Nous voyons que le jeu le plus joué est **Wii Sports** avec un total de **82,74 millions de dollars** dans le monde.


Nous tracerons ce qui précède à l'aide d'un graphique circulaire.

In [20]:
pie1 = px.pie(top, values=top['Global_Sales'][:10], names=top['Name'][:10],title='Top 10 games globally', 
              color_discrete_sequence=px.colors.sequential.Purp_r)
pie1.update_traces(textposition='inside', textinfo='percent+label',showlegend=False)

pie1.show()

Le graphique à secteurs montre également la proportion des ventes de chaque jeu, tout en décrivant les résultats.

# 4. Quels sont les meilleurs jeux pour différentes régions ?

* Nous devrons comparer les ventes réalisées par différents jeux au niveau régional.

In [21]:
name2 = pd.DataFrame(df.groupby("Name")[["NA_Sales"]].mean().sort_values(by=['NA_Sales'],ascending=[False]).reset_index())
name2.rename(columns = {'Name':'Name_NA'}, inplace = True)

name3 = pd.DataFrame(df.groupby("Name")[["EU_Sales"]].mean().sort_values(by=['EU_Sales'],ascending=[False]).reset_index())
name3.rename(columns = {'Name':'Name_EU'}, inplace = True)

name4 = pd.DataFrame(df.groupby("Name")[["JP_Sales"]].mean().sort_values(by=['JP_Sales'],ascending=[False]).reset_index())
name4.rename(columns = {'Name':'Name_JP'}, inplace = True)

name5 = pd.DataFrame(df.groupby("Name")[["Other_Sales"]].mean().sort_values(by=['Other_Sales'],ascending=[False]).reset_index())
name5.rename(columns = {'Name':'Name_other'}, inplace = True)

#Concatenating the results.
name_df=pd.concat([name2,name3,name4,name5],axis=1)

In [22]:
subplot_name1 = make_subplots(rows=4, cols=1, shared_yaxes=True,subplot_titles=("North American top games","Europe top games", "Japan top games","Other regions top games",'Top games globally'))

#Subplot for North America
subplot_name1.add_trace(go.Bar(x=name_df['Name_NA'][:5], y=name_df['NA_Sales'][:5],marker=dict(color=[1, 2, 3],coloraxis="coloraxis")),1, 1)

#Subplot for Europe
subplot_name1.add_trace(go.Bar(x=name_df['Name_EU'][:5], y=name_df['EU_Sales'][:5],marker=dict(color=[4, 5, 6], coloraxis="coloraxis")), 2, 1)

#Subplot for Japan
subplot_name1.add_trace(go.Bar(x=name_df['Name_JP'][:5], y=name_df['JP_Sales'][:5],marker=dict(color=[7, 8, 9], coloraxis="coloraxis")),3, 1)

#Subplot for other regions
subplot_name1.add_trace(go.Bar(x=name_df['Name_other'][:5], y=name_df['Other_Sales'][:5],marker=dict(color=[10, 11, 12], coloraxis="coloraxis")),4, 1)

subplot_name1.update_layout(height=1000,width=500,coloraxis=dict(colorscale='Mint_r'), showlegend=False)
subplot_name1.update_xaxes(tickangle=45)
subplot_name1.show()

Le graphique nous montre les meilleurs jeux préférés par les utilisateurs dans différentes régions et également dans le monde. Nous observons ce qui suit :

* **Wii Sports** a été le meilleur jeu en Amérique du Nord, en Europe et dans d'autres régions.
* **Pokemon Red/Pokemon Blue** est le meilleur jeu au Japon.

# Conclusion
D'après les données ci-dessus, nous pouvons dire que les jeux d'action sur DC ou Playstation sont les plus populaires et sont ceux qui génèrent le maximum de ventes dans le monde entier. De plus, comme ces jeux sont si abondants et populaires (en termes de classement), la variation des ventes d'un ou deux de ces jeux n'entraînerait pas de changement significatif dans les ventes globales.##